types/aframe/index.d.ts

```diff
@@ -0,0 +1,324 @@
+declare var hasNativeWebVRImplementation: boolean;
+
+interface NodeSelector {
+	querySelector(selectors: 'a-scene'): AFrame.Scene;
+	querySelector<T extends AFrame.Entity<any>>(selectors: string): T;
+	querySelectorAll(selectors: string): NodeListOf<AFrame.Entity<any> | Element>;
+}
+
+interface Document {
+	createElement(tagName: string): AFrame.Entity;
+}
+
+// Interfaces
+declare namespace AFrame {
+	interface ObjectMap<T = any> {
+		[ key: string ]: T;
+	}
+
+	interface AFrameGlobal {
+		AEntity: Entity;
+		ANode: ANode;
+		AScene: Scene;
+		components: { [ key: string ]: ComponentDescriptor };
+		geometries: { [ key: string ]: GeometryDescriptor };
+		primitives: { [ key: string ]: Entity };
+		registerComponent(name: string, component: ComponentDefinition): ComponentConstructor;
+		registerElement(name: string, element: ANode): void;
+		registerGeometry(name: string, geometery: THREE.Geometry): Geometry;
+		registerPrimitive(name: string, primitive: PrimitiveDefinition): void;
+		registerShader(name: string, shader: any): void;
+		registerSystem(name: string, definition: SystemDefinition): void;
+		schema: SchemaUtils;
+		shaders: { [ key: string ]: ShaderDescriptor };
+		systems: { [key: string]: System };
+		THREE: any; // TODO THREE
+		TWEEN: any; // TODO TWEEN
+		utils: Utils;
+		version: string;
+	}
+
+	interface Animation {
+		attribute: string;
+		begin: string | number;
+		delay: number;
+		direction: 'alternate' | 'alternateReverse' | 'normal' | 'reverse';
+		dur: number;
+		easing(): void;
+		end: string;
+		fill: 'backwards' | 'both' | 'forwards' | 'none';
+		from: any; // TODO type
+		repeat: number | 'indefinite';
+		to: number;
+	}
+
+	interface ANode extends HTMLElement {
+		// Only public APIs added. Many methods intentionally left out.
+		// createdCallback
+		// attachedCallback
+		// attributeChangedCallback
+		closestScene(): Scene;
+		closest(selector: string): ANode;
+		// detachedCallback
+		hasLoaded: boolean;
+		load(cb?: () => void, childFilter?: (el: Element) => boolean): void;
+		// updateMixins
+		registerMixin(id: string): void;
+		setAttribute(type: string, newValue: any): void;
+		unregisterMixin(id: string): void;
+		removeMixinListener(id: string): void;
+		attachMixinListener(mixin: HTMLElement): void;
+		emit(name: string, detail?: any, bubbles?: boolean): void;
+		emitter(name: string, detail?: any, bubbles?: boolean): () => void;
+	}
+
+	interface Behavior {
+		tick(): void;
+	}
+
+	interface Component {
+		attrName?: string;
+		data?: any;
+		dependencies?: string[];
+		el: Entity;
+		id: string;
+		multiple?: boolean;
+		name: string;
+		schema: Schema;
+
+		init(): void;
+		pause(): void;
+		play(): void;
+		remove(): void;
+		tick?(time: number, timeDelta: number): void;
+		update(oldData: any): void;
+		updateSchema?(): void;
+
+		extendSchema(update: Schema): void;
+		flushToDOM(): void;
+	}
+
+	interface ComponentConstructor {
+		new (el: Entity, name: string, id: string): Component;
```

Comment:
Are you sure that this Isn't a [mistake](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes)?